lxc/completion.go

```diff
@@ -372,8 +372,9 @@ func (g *cmdGlobal) cmpInstanceSetKeys(instanceName string) ([]string, cobra.She
 
 	for configKey := range instance.Config {
 		// We only want to return the intersection between allInstanceConfigKeys and configKeys to avoid returning the full instance config.
+		// We also need to account for user defined keys such as user.* and linux.sysctl.*.
 		_, exists := keySet[configKey]
-		if exists {
+		if exists || shared.StringHasPrefix(configKey, "user.", "linux.sysctl.", "environment.") {
```

Comment:
Shouldnt we make this a slice of prefixes somewhere so that both the lxc client and the lxd server can use it when validating keys so we can see the references ? Otherwise how will we remember to update this list when a new prefix is added?


For example, it looks like there's some other prefix keys not accounted for

limits.kernel.
image.
environment.

See `ConfigKeyChecker()`

Reply:
Good call, I'll add the slices to `lxd/instance/instancetype`.